osu.Game/Utils/ModUtils.cs

```diff
@@ -229,6 +229,38 @@ public static bool InstantiateValidModsForRuleset(Ruleset ruleset, IEnumerable<A
             return proposedWereValid;
         }
 
+        /// <summary>
+        /// Verifies all mods provided belong to the given ruleset.
+        /// </summary>
+        /// <param name="ruleset">The ruleset to check the proposed mods against.</param>
+        /// <param name="proposedMods">The mods proposed for checking.</param>
+        /// <returns>Whether all <paramref name="proposedMods"/> belong to the given <paramref name="ruleset"/>.</returns>
+        public static bool CheckModsBelongToRuleset(Ruleset ruleset, IEnumerable<Mod> proposedMods)
+        {
+            var rulesetModsTypes = ruleset.AllMods.Select(m => m.GetType()).ToList();
+
+            foreach (var proposedMod in proposedMods)
+            {
+                bool found = false;
+
+                var proposedModType = proposedMod.GetType();
+
+                foreach (var rulesetModType in rulesetModsTypes)
+                {
+                    if (rulesetModType.IsAssignableFrom(proposedModType))
```

Comment:
Can you write a test for this helper because every time I see `IsAssignableFrom()` my mind instantly shuts down.

Reply:
Yeah that is a funny one. We could simply change to use `IsAssignableTo` everywhere (which is simply `IsAssignableFrom` with opposite sides under the hood), but I will leave that thought for a rainy day.